packages/vscode-extension/src/commands/UploadFile.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ ** Copyright (c) 2020 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+
+import * as path from 'path';
+import { window, workspace, InputBoxOptions } from 'vscode';
+import SuiteCloudRunner from '../core/SuiteCloudRunner';
+import MessageService from '../service/MessageService';
+import { COMMAND, UPLOAD_FILE, YES, NO } from '../service/TranslationKeys';
+import { actionResultStatus, CLIConfigurationService, ApplicationConstants, FileCabinetService } from '../util/ExtensionUtil';
+import BaseAction from './BaseAction';
+
+export default class UploadFile extends BaseAction {
+	readonly commandName: string = 'file:upload';
+
+	async execute(opts: { suiteCloudRunner: SuiteCloudRunner; messageService: MessageService }) {
+		const activeFile = window.activeTextEditor?.document.uri;
+		if (!activeFile) {
+			opts.messageService.showCommandError(this.translationService.getMessage(UPLOAD_FILE.ERRORS.INVALID_ACTIVE_FILE));
+			return;
+		}
+		const workspaceFolder = workspace.getWorkspaceFolder(activeFile);
+		if (!workspaceFolder) {
+			opts.messageService.showCommandError(this.translationService.getMessage(UPLOAD_FILE.ERRORS.INVALID_WORKSPACE));
+			return;
+		}
```

Comment:
I think these checks are already covered in the ActionExecutor in line 20-21 (when checking for the executionPath) that is a kind of precondition to execute any type of action.  But probably the messages here are more accurate. Maybe we can move them to these precondition check in the ActionExecutor?